chrome/content/zotero/feedSettings.js

```diff
@@ -0,0 +1,188 @@
+/*
+    ***** BEGIN LICENSE BLOCK *****
+    
+    Copyright © 2015 Center for History and New Media
+                     George Mason University, Fairfax, Virginia, USA
+                     http://zotero.org
+    
+    This file is part of Zotero.
+    
+    Zotero is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    
+    Zotero is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+    
+    You should have received a copy of the GNU Affero General Public License
+    along with Zotero.  If not, see <http://www.gnu.org/licenses/>.
+    
+    ***** END LICENSE BLOCK *****
+*/
+
+//////////////////////////////////////////////////////////////////////////////
+//
+// Zotero_Feed_Settings
+//
+//////////////////////////////////////////////////////////////////////////////
+
+var Zotero_Feed_Settings = new function() {
+	let urlIsValid = true,
+		data = null,
+		feedReader = null,
+		urlTainted = false;
+	
+	let cleanURL = function(url) {
+		url = url.trim();
+		if (!url) return;
+		
+		let ios = Components.classes["@mozilla.org/network/io-service;1"]
+			.getService(Components.interfaces.nsIIOService);
+		
+		let cleanUrl;
+		try {
+			let uri = ios.newURI(url, null, null);
+			if (uri.scheme != 'http' && uri.scheme != 'https') {
+				Zotero.debug(uri.scheme + " is not a supported protocol for feeds.");
+			}
+			
+			cleanUrl = uri.spec;
+		} catch (e) {
+			if (e.result == Components.results.NS_ERROR_MALFORMED_URI) {
+				// Assume it's a URL missing "http://" part
+				try {
+					cleanUrl = ios.newURI('http://' + url, null, null).spec;
+				} catch (e) {}
+			}
+			throw e;
+		}
+		
+		if (!cleanUrl) return;
+		
+		if (/^https?:\/\/[^\/\s]+\/\S/.test(cleanUrl)) return cleanUrl;
+	};
+	
+	this.init = function() {
+		this.toggleAdvancedOptions(false);
+		
+		data = window.arguments[0];
+		
+		if (data.url) {
+			document.getElementById('feed-url').value = data.url;
+		}
+		
+		if (!data.url) {
+			this.invalidateUrl();
+		} else {
+			// Do not allow to change URL for existing feed
+			document.getElementById('feed-url').readOnly = true;
+		}
+		
+		if (data.title) {
+			document.getElementById('feed-title').value = data.title;
+		}
+		
+		let ttl;
+		if (data.ttl !== undefined) {
+			ttl = Math.floor(data.ttl / 60);
+		} else {
+			ttl = 1;
+		}
+		document.getElementById('feed-ttl').value = ttl;
```

Comment:
Not sure why I originally thought this was a good idea, but I think that if users can only set the TTL in hour increments, we should store it as such (rather than minutes). Maybe there was a good reason though.